src/TwigYard/Component/ConfigCache.php

```diff
@@ -63,18 +76,38 @@ public function getConfig(string $sitesDir, string $siteConfig): array
                         $logger->error($e->getMessage());
                         continue;
                     }
-                    if (!isset($config['url']['canonical']) || isset($configs[$config['url']['canonical']])) {
+
+                    if (isset($config['parameters'])) {
+                        $paramBag = new ParameterBag($config['parameters']);
+                        $config = $paramBag->resolveValue($config);
+                    }
+
+                    if (!isset($config['version']) || $config['version'] === 1) {
+                        if (!isset($config['url'])) {
+                            continue;
+                        }
+                        $urlConf = $config['url'];
+                        $config['version'] = 1;
+                    } elseif ($config['version'] === 2) {
+                        if (!isset($config['middlewares']['url'])) {
+                            continue;
+                        }
+                        $urlConf = $config['middlewares']['url'];
+                    }
```

Comment:
I would like to add `else` block and throw InvalidSiteConfigException if unsupported version is used

Reply:
I think it is needed as the config is ignored if `$urlConf` is not set. I added a test for it though.